charts/camunda-platform/templates/web-modeler/_helpers.tpl

```diff
@@ -200,14 +200,60 @@ Define match labels for Web Modeler websockets to be used in matchLabels selecto
 [web-modeler] Get the name of the secret that contains the database password, depending on whether the postgresql dependency chart is enabled.
 */}}
 {{- define "webModeler.restapi.databaseSecretName" -}}
-  {{- .Values.postgresql.enabled | ternary (include "webModeler.postgresql.fullname" .) (include "webModeler.restapi.fullname" .) -}}
+  {{- if .Values.postgresql.enabled }}
+    {{- .Values.postgresql.auth.existingSecret | default (include "webModeler.postgresql.fullname" .) }}
+  {{- else }}
+    {{- if or (typeIs "string" .Values.webModeler.restapi.externalDatabase.existingSecret) .Values.webModeler.restapi.externalDatabase.password }}
+      {{- include "webModeler.restapi.fullname" . }}
+    {{- else if typeIs "map[string]interface {}" .Values.webModeler.restapi.externalDatabase.existingSecret }}
+      {{- .Values.webModeler.restapi.externalDatabase.existingSecret.name | default (include "webModeler.restapi.fullname" .) }}
+    {{- end }}
```

Comment:
I noticed that you switched the order of the if clauses. It probably doesn't matter too much, but the logic doesn't fit the comment on `webModeler.restapi.externalDatabase.password` any more now:

> ignored if `webModeler.restapi.externalDatabase.existingSecret` is set

If you provided both `password` and `existingSecret.name`, the `password` would be used (and not be ignored).

The same holds true for `webModeler.restapi.mail.smtpPassword`.

Reply:
According to this snippet,
https://github.com/camunda/camunda-platform-helm/blob/cda20c295d7f0a0ccb0c883bf245beac7aa07538/charts/camunda-platform/templates/web-modeler/secret-restapi.yaml#L13-L15

If existingSecret is set, the password field is not used.  So the logic in that comment refers to the `secret-restapi.yaml` and not inside this helpers function.

Reply:
I quickly tried this and provided both `externalDatabase.existingSecret.name` and `externalDatabase.password`. You'll end up with the following error when you run `helm install`:

```
Error: INSTALLATION FAILED: template: camunda-platform/templates/web-modeler/secret-restapi.yaml:14:148: executing "camunda-platform/templates/web-modeler/secret-restapi.yaml" at <b64enc>: wrong type for value; expected string; got map[string]interface {}
```

But it would be a misconfiguration by the user anyway (you should either set one or the other), so I'd say it's acceptable :).

Reply:
I'm inclined to leave this one. I wouldn't want to add more complicated logic to cleverly handle this.